### PR TITLE
Jd 2023 10 file upload

### PR DESCRIPTION
**Pull request by goapunk**
based on https://github.com/liqd/ckeditor5-file-uploader

### CODE REVIEW

File: django_ckeditor_5/forms.py [hunk 1]
```diff
@@ -1,5 +1,10 @@
 from django import forms
+from django.core.validators import FileExtensionValidator
+from django.conf import settings
 
 
 class UploadFileForm(forms.Form):
-    upload = forms.FileField()
+    upload = forms.FileField(validators=[
+        FileExtensionValidator(getattr(settings, "CKEDITOR_5_UPLOAD_FILE_TYPES", ["jpeg", "png", "gif", "bmp", "webp", "tiff"]))
```
Comment (m4ra) [inline django_ckeditor_5/forms.py:8]: why there are more file types here than in the settings?

File: django_ckeditor_5/views.py [hunk 2]
```diff
@@ -52,20 +54,26 @@ def image_verify(f):
         raise NoImageException
 
 
-def handle_uploaded_file(f):
+def handle_uploaded_file(user, f):
     fs = storage()
-    filename = fs.save(f.name, f)
+    path = f.name
+    if getattr(settings, "CKEDITOR_5_PATH_FROM_USERNAME", False):
+        path = storage.get_available_name(os.path.join(user.username, f.name))
```
Comment (m4ra) [inline django_ckeditor_5/views.py:61]: i know u have been working on this long, but it would be nice to switch using pathlib instead of os.path.

Comment (goapunk) [reply django_ckeditor_5/views.py]: no worries, if there are things which should be changed they should be changed!

File: django_ckeditor_5/views.py [hunk 3]
```diff
@@ -52,20 +54,26 @@ def image_verify(f):
         raise NoImageException
 
 
-def handle_uploaded_file(f):
+def handle_uploaded_file(user, f):
     fs = storage()
-    filename = fs.save(f.name, f)
+    path = f.name
+    if getattr(settings, "CKEDITOR_5_PATH_FROM_USERNAME", False):
+        path = storage.get_available_name(pathlib.PurePath(user.username, f.name))
```
Comment (goapunk) [inline django_ckeditor_5/views.py:61]: I went it `PurePath` as all the verification is done by `get_available_name` anyways

File: django_ckeditor_5/static/django_ckeditor_5/src/ckeditor.js [hunk 4]
```diff
@@ -38,19 +38,20 @@ import ListProperties from '@ckeditor/ckeditor5-list/src/listproperties';
 import SourceEditing from '@ckeditor/ckeditor5-source-editing/src/sourceediting';
 import GeneralHtmlSupport from '@ckeditor/ckeditor5-html-support/src/generalhtmlsupport';
 import ImageInsert from '@ckeditor/ckeditor5-image/src/imageinsert';
-import {TableCaption} from "@ckeditor/ckeditor5-table";
+import {TableCaption} from '@ckeditor/ckeditor5-table';
```
[the remark this answers is not in the document]
Comment (goapunk) [reply django_ckeditor_5/static/django_ckeditor_5/src/ckeditor.js:41]: @hom3mad3 you can test it via the meinberlin pr (there are some instructions there: https://github.com/liqd/a4-meinberlin/pull/5260)

**Review by philli-m — APPROVED** [review]
looking good from the parts I recognise and it works so far so lets merge and see what the testers find

**Pull request merged**